vulkaninfo/vulkaninfo.cpp

```diff
@@ -756,7 +756,18 @@ int main(int argc, char **argv) {
     }
 
     if (selected_gpu >= gpus.size()) {
-        selected_gpu = 0;
+        std::cout << "The selected gpu (" << selected_gpu << ") is invalid. Please select a valid gpu.\n";
+        std::cout << "Valid gpus are the following: ";
```

Comment:
It's a range, correct?  No need for a list, how about:
"GPU is not in the valid range of 0 to X"

Reply:
I did that initially. But "valid range of 0 to 0" looks very silly. At first I handled the 1 gpu case seperately, but the text was awkward.

Reply:
The main point is to give a clue on a multi-gpu system, what the app expects for selection; e.g.: is it numeric or string, an index or some device ID, is it zero-based or one-based, etc.  If they specify "1" on a "range of 0 to 0", the user probably gets the hint.  It's a bit silly to specify an index on a single-gpu system, wouldn't worry about awkwardness of the message

Just give a quick verbal description of the argument check that failed, with enough info that so user can fix.

Reply:
dont spend too much time on this.  The important thing is to convey info.  dont need to say "please supply a valid gpu", they already know they've given in invalid answer.

Reply:
I see your point. Amended and simplified it.